test/queryBuilders.js

```diff
@@ -0,0 +1,219 @@
+const t = require('../test-lib/test.js');
+const assert = require('assert');
+
+describe('Query Builders', function() {
+  this.timeout(t.timeout);
+
+  let apos;
+  after(function() {
+    return t.destroy(apos);
+  });
+
+  before(async function () {
+    apos = await t.create({
+      root: module,
+      modules: {
+        young: {
+          options: {
+            alias: 'young'
+          },
+          extend: '@apostrophecms/piece-type',
+          fields: {
+            add: {
+              age: {
+                label: 'Age',
+                type: 'integer',
+                required: true
+              }
+            }
+          },
+          queries(self, query) {
+            return {
+              builders: {
+                age: {
+                  launder(str) {
+                    return [ 'children', 'adult' ].includes(str) ? str : null;
+                  },
+                  finalize() {
+                    const age = query.get('age');
+
+                    if ([ 'children', 'adults' ].includes(age)) {
+                      const ageCriteria = age === 'children' ? { $lte: 18 } : { $gt: 18 };
+                      query.and({ age: ageCriteria });
+                    }
+                  }
+                }
+              },
+              methods: {
+                async sortByAge() {
+                  await query.finalize();
+
+                  const pipeline = [
+                    { $match: query.get('criteria') },
+                    { $sort: { age: 1 } }
+                  ];
+
+                  const results = await self.apos.doc.db.aggregate(pipeline).toArray();
+
+                  return results;
+                }
+              }
+            };
+          }
+        },
+        person: {
+          extend: 'young',
+          options: {
+            alias: 'person'
+          },
+          extendQueries(self, query) {
+            return {
+              builders: {
+                age: {
+                  def: 'adult',
+                  launder(_super, str) {
+                    return _super() || str === 'senior';
+                  },
+                  finalize(_super) {
+                    _super();
```

Comment:
finalize is async and should be awaited. I know your examples aren't so you get away with it, but I don't want to give anyone the wrong idea about finalize.

Reply:
e.g. if someone uses this as a guide to extending an existing finalize function they may get a surprise if they don't await. Always safe to make your subclass version async just in case the super becomes async.

Reply:
This is just a concern about the tests. The code is fine.

Reply:
Makes sense, I modify that.